src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+import java.util.ArrayList;
+/**
+ * Class responsible for handling the logic of the task lists.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList = new ArrayList<>();
+    /**
+     * Create a task list based on the contents.
+     * @param contents tasks to be added to task list
+     */
+    public TaskList(ArrayList<String> contents) {
+        for (String c : contents) {
```

Comment:
Variable c can be written more clearer, but I think since it is only used here so should be fine.